eclipse-collections/src/main/java/org/eclipse/collections/impl/string/immutable/CharAdapter.java

```diff
@@ -531,6 +537,38 @@ public int hashCode()
         return hashCode;
     }
 
+    /**
+     * @since 9.1.
+     */
+    @Override
+    public ImmutableList<CharCharPair> zipChar(CharList list)
+    {
+        MutableList<CharCharPair> target = Lists.mutable.empty();
+        int size = this.size();
+        int othersize = list.size();
+        for (int i = 0; i < size && i < othersize; i++)
+        {
+            target.add(PrimitiveTuples.pair(this.get(i), list.get(i)));
+        }
+        return target.toImmutable();
+    }
+
+    /**
+     * @since 9.1.
+     */
+    @Override
+    public <T> ImmutableList<CharObjectPair<T>> zip(ListIterable<T> list)
```

Comment:
Should the scope of T be expanded `<T extends ?>` ?

Reply:
Not sure what you are suggesting in this case.  This is consistent with definition in ListIterable.

```    
    @Override
    <S> ListIterable<Pair<T, S>> zip(Iterable<S> that);
```

Reply:
ok makes sense.